Improve List demo description - grouped list (#556)
DevExpress/devextreme-demos: JSDemos/Demos/List/GroupedList/description.md

> It is possible to display the **List** in groups under the following conditions:

Reviewer: "display the list in groups" is a vague statement. Data Grid can display data in groups - and there can be unlimited number of hierarchy levels. I think we need to be specific here that we're strictly speaking about two-level hierarchies. It's a simple parent-child relationship. 

I also think it's best to restructure the information a little. We can mention the property right away and then we can clearly separate two ways to supply grouped data (using sub-headers). 

```suggestion
**List** can display items as a two-level hierarchy - parent and child items. To group items, set the [grouped](/Documentation/ApiReference/UI_Components/dxList/Configuration/#grouped) property to **true** and make sure that the data source defines the hierarchy in one of the following ways. 
```

> - The **List** data passed to the **dataSource** is structured so that each group is an object that contains two fields: **key** and **items**. The **key** field represents a group header, and the **items** field represents an array of group items.

Reviewer: As I mentioned above, I want to use sub-headers to clearly differentiate different methods to supply data. And here I want to have all information regarding data sources that have items already grouped - whether or not field mapping is used. (In your version these things were separated for some reason.)

```suggestion
### Hierarchical Data Source Structure

You can use a **dataSource** where each entry is an object that contains two fields: **key** and **items**. The **key** field specifies the group header (parent), and the **items** field holds an array of child items. This demo shows how to use such a data source.

A data source can specify different field names as long as the data is structured in the same manner. If that's the case, implement the DataSource's [map](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#map) function to create **key** and **items** field mappings. Refer to the following help topic for more information: [Item Mapping](/Documentation/Guide/Data_Binding/Data_Layer/#Reading_Data/Data_Transformation/Item_Mapping).
```

>     In cases when the data source doesn't conform to the structure specified above, use the [DataSource](https://js.devexpress.com/Documentation/ApiReference/Data_Layer/DataSource/)'s [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) property to define the data field to group against. If data objects are grouped but use other field names, implement the DataSource's [map](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#map) function to create **key** + **items** field mappings. Refer to the following help topic for more information: [Item Mapping](/Documentation/Guide/Data_Binding/Data_Layer/#Reading_Data/Data_Transformation/Item_Mapping).

Reviewer: Now switching to regular data sources where we can specify a grouping field. But note below - I only assume what the list will display if we use such a data source. You'll need to fill in the blanks. 

```suggestion
### Grouping Field

The data source can contain a plain list of objects with multiple fields. Assign a field to the [DataSource](https://js.devexpress.com/Documentation/ApiReference/Data_Layer/DataSource/)'s [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) property to create a hierarchical display. Parent items will display unique values from the specified field. Child items will display grouped values. 
```

Author: I'm not quite clear on what are you describing here.
Are you talking about the case when the data looks like so:
``` javascript
var data = [
    { name: "Amelia", birthYear: 1991, gender: "female" },
    { name: "Benjamin", birthYear: 1983, gender: "male" },
    { name: "Andrew", birthYear: 1991, gender: "male" },
    { name: "Daniela", birthYear: 1983, gender: "female" },
    { name: "Lee", birthYear: 1983, gender: "male" },
    { name: "Betty", birthYear: 1983, gender: "female" }
];
var dataSource = new DevExpress.data.DataSource(data);
```

Basically, are you talking about the scenario where we use dataSource's group() method?
``` javascript
dataSource.group("birthYear");
dataSource.load().done(function(result) {
    //'result' contains the loaded array
});
```

to create a hierarchy like so:

```javascript
[
    {
        key: 1983,
        items: [
            { name: "Benjamin", birthYear: 1983, gender: "male" },
            { name: "Daniela", birthYear: 1983, gender: "female" },
            { name: "Lee", birthYear: 1983, gender: "male" },
            { name: "Betty", birthYear: 1983, gender: "female" }
        ]
    },
    {
        key: 1991,
        items: [
            { name: "Amelia", birthYear: 1991, gender: "female" },
            { name: "Andrew", birthYear: 1991, gender: "male" }
        ]
    }
]
```

Reviewer: Yes, that's what I thought you meant by "use the [DataSource]'s [group] property to define the data field to group against."

If that group property doesn't work like that, then how does it work?

> - The [grouped](/Documentation/ApiReference/UI_Components/dxList/Configuration/#grouped) property is set to **true**.    

Reviewer: I moved that to the beginning, so it's no longer necessary here.

> - The **List** data passed to the **dataSource** is structured so that each group is an object that contains two fields: **key** and **items**. The **key** field represents a group header, and the **items** field represents an array of group items. Refer to this [help guide](/Documentation/Guide/UI_Components/List/Grouping/In_the_Data_Source/) for more information on grouping in Data Source.   
> 
> In cases when the data source doesn't conform to the structure specified above, use the [DataSource](https://js.devexpress.com/Documentation/ApiReference/Data_Layer/DataSource/)'s [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) property to define the data field to group against. If data objects are grouped but use other field names, implement the DataSource's [map](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#map) function to create **key** + **items** field mappings. Refer to this [help guide](/Documentation/Guide/Data_Binding/Data_Layer/#Reading_Data/Data_Transformation/Item_Mapping) for more information on item mapping. 

Reviewer 2: I suggest removing the link to the Grouping in the Data Source article because it doesn't add anything to the things we say in this description. Also, I changed the last sentence a bit - it is better to give the name of the article you give a link to instead of just "this article".

```suggestion
- The **List** data passed to the **dataSource** is structured so that each group is an object that contains two fields: **key** and **items**. The **key** field represents a group header, and the **items** field represents an array of group items.

    In cases when the data source doesn't conform to the structure specified above, use the [DataSource](https://js.devexpress.com/Documentation/ApiReference/Data_Layer/DataSource/)'s [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) property to define the data field to group against. If data objects are grouped but use other field names, implement the DataSource's [map](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#map) function to create **key** + **items** field mappings. Refer to the following help topic for more information: [Item Mapping](/Documentation/Guide/Data_Binding/Data_Layer/#Reading_Data/Data_Transformation/Item_Mapping).
```

> - The [grouped](/Documentation/ApiReference/UI_Components/dxList/Configuration/#grouped) property is set to *true*.    

Reviewer 2: ```suggestion
- The [grouped](/Documentation/ApiReference/UI_Components/dxList/Configuration/#grouped) property is set to **true**.    
```

> If you need to make the groups collapsible, set the [collapsibleGroups](/Documentation/ApiReference/UI_Components/dxList/Configuration/#collapsibleGroups) to **true**. 

Reviewer: Let's add another sub-header for these things - e.g. "Group Display Settings/Customization"

> If you need to make the groups collapsible, set the [collapsibleGroups](/Documentation/ApiReference/UI_Components/dxList/Configuration/#collapsibleGroups) to *true*. 

Reviewer 2: ```suggestion
If you need to make the groups collapsible, set the [collapsibleGroups](/Documentation/ApiReference/UI_Components/dxList/Configuration/#collapsibleGroups) to **true**. 
```

> You can also [customize](/Documentation/Guide/UI_Components/List/Grouping/Customize_Group_Headers/) the look of group headers. To do so, use the [groupTemplate](/Documentation/ApiReference/UI_Components/dxList/Configuration/#groupTemplate) property.

Reviewer 2: Same here, the article is not very useful if you already looked at the demo. A link to the API reference is sufficient.

```suggestion
You can also customize the look of group headers. To do so, use the [groupTemplate](/Documentation/ApiReference/UI_Components/dxList/Configuration/#groupTemplate) property.
```

> **List** can display items as a two-level hierarchy - parent and child items. To group items, set the [grouped](/Documentation/ApiReference/UI_Components/dxList/Configuration/#grouped) property to **true** and make sure that the data source defines the hierarchy in one of the following ways. 

Reviewer 3: What is "**List**"? List what? The List component?

```suggestion
The **List** component can display items as a two-level hierarchy: parent and child items. To group items, set the [grouped](/Documentation/ApiReference/UI_Components/dxList/Configuration/#grouped) property to **true** and ensure the data source defines the hierarchy in one of the following ways:
```

### Hierarchical Data Source Structure

> You can use a [dataSource](/Documentation/ApiReference/UI_Components/dxList/Configuration/#dataSource) where each entry is an object that contains two fields: **key** and **items**. The **key** field specifies the group header (parent), and the **items** field holds an array of child items. This demo shows how to use such a data source.

Reviewer 3: ```suggestion
You can use a [dataSource](/Documentation/ApiReference/UI_Components/dxList/Configuration/#dataSource) where each entry is an object that contains the **key** and **items** fields. The **key** field specifies the group header (parent), and the **items** field holds an array of child items. This demo shows how to use this type of data source.
```

> A data source can specify different field names as long as the data is structured in the same manner. If that's the case, implement the [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/)'s [map](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#map) function to create **key** and **items** field mappings. Refer to the following help topic for more information: [Item Mapping](/Documentation/Guide/Data_Binding/Data_Layer/#Reading_Data/Data_Transformation/Item_Mapping).

Reviewer 3: ```suggestion
A data source can specify different field names if the data is structured in the same way. In this case, implement the [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/)'s [map](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#map) function to create **key** and **items** field mappings. Refer to the following help topic for more information: [Item Mapping](/Documentation/Guide/Data_Binding/Data_Layer/#Reading_Data/Data_Transformation/Item_Mapping).
```

### Grouping Field

> The data source can contain a plain list of objects with multiple fields. Assign a field to the DataSource's [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) property to create a hierarchical display. Parent items will display unique values from the specified field. Child items will display grouped values. 

Reviewer 3: ```suggestion
The data source can contain a plain list of objects with multiple fields. Assign a field to the DataSource's [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) property to create a hierarchical display. Parent items display unique values from the specified field, and Child items display grouped values. 
```

### Group Display Settings

If you need to make the groups collapsible, set the [collapsibleGroups](/Documentation/ApiReference/UI_Components/dxList/Configuration/#collapsibleGroups) to **true**. 

> You can also customize the look of group headers. To do so, use the [groupTemplate](/Documentation/ApiReference/UI_Components/dxList/Configuration/#groupTemplate) property.

Reviewer 3: ```suggestion
You can also use the [groupTemplate](/Documentation/ApiReference/UI_Components/dxList/Configuration/#groupTemplate) property to customize the appearance of group headers.
```

**List** can display items as a two-level hierarchy - parent and child items. To group items, set the [grouped](/Documentation/ApiReference/UI_Components/dxList/Configuration/#grouped) property to **true** and make sure that the data source defines the hierarchy in one of the following ways. 

### Hierarchical Data Source Structure

> You can use a **dataSource** where each entry is an object that contains two fields: **key** and **items**. The **key** field specifies the group header (parent), and the **items** field holds an array of child items. This demo shows how to use such a data source.

Reviewer 2: ```suggestion
You can use a [dataSource](/Documentation/ApiReference/UI_Components/dxList/Configuration/#dataSource) where each entry is an object that contains two fields: **key** and **items**. The **key** field specifies the group header (parent), and the **items** field holds an array of child items. This demo shows how to use such a data source.
```

> A data source can specify different field names as long as the data is structured in the same manner. If that's the case, implement the DataSource's [map](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#map) function to create **key** and **items** field mappings. Refer to the following help topic for more information: [Item Mapping](/Documentation/Guide/Data_Binding/Data_Layer/#Reading_Data/Data_Transformation/Item_Mapping).

Reviewer 2: ```suggestion
A data source can specify different field names as long as the data is structured in the same manner. If that's the case, implement the [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/)'s [map](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#map) function to create **key** and **items** field mappings. Refer to the following help topic for more information: [Item Mapping](/Documentation/Guide/Data_Binding/Data_Layer/#Reading_Data/Data_Transformation/Item_Mapping).
```

### Grouping Field

> The data source can contain a plain list of objects with multiple fields. Assign a field to the [DataSource](https://js.devexpress.com/Documentation/ApiReference/Data_Layer/DataSource/)'s [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) property to create a hierarchical display. Parent items will display unique values from the specified field. Child items will display grouped values. 

Reviewer 2: ```suggestion
The data source can contain a plain list of objects with multiple fields. Assign a field to the DataSource's [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) property to create a hierarchical display. Parent items will display unique values from the specified field. Child items will display grouped values. 
```

### Group Display Settings

> 
> - The [grouped](/Documentation/ApiReference/UI_Components/dxList/Configuration/#grouped) property is set to **true**.    

Reviewer 2: ```suggestion
```
